src/IceRpc.Slice.Generators/Internal/ServiceClass.cs

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) ZeroC, Inc.
+
+namespace IceRpc.Slice.Generators.Internal;
+
+/// <summary>Represents a C# class that has the <c>IceRpc.Slice.SliceServiceAttribute</c> attribute and for which this
+/// generator implements the <c>IceRpc.IDispatcher</c> interface.</summary>
+internal class ServiceClass : ContainerDefinition
+{
+    /// <summary>Gets a value indicating whether the service has a base service class.</summary>
+    internal bool HasBaseServiceClass { get; }
+
+    /// <summary>Gets the C# namespace containing this definition.</summary>
```

Comment:
Is this the namespace or the scope?

This service class can presumably be nested in some other class or struct.

Reply:
> Is this the namespace or the scope?

This contains both, mimics the nesting of the partial definition with the SliceService attribute

Reply:
So it should be scope? For me, scope corresponds to both namespace(s) and enclosing structs/classes.

Reply:
Renamed as "Scope"

Reply:
I was mistaken here, this is just the namespace and it can be null

See https://learn.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.isymbol.containingnamespace?view=roslyn-dotnet-4.6.0